Caching/Sources/Caching/Caching.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Caching.swift
+//
+//
+//  Created by ziad on 11/08/2022.
+//
+
+import UIKit
+
+extension Encodable {
+    
+    func asDictionary() throws -> [String: Any] {
+        let data = try JSONEncoder().encode(self)
+        guard let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any] else {
+            throw NSError()
+        }
+        return dictionary
+    }
+
+    func asArray() throws -> [[String: Any]] {
+        let data = try JSONEncoder().encode(self)
+        guard let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [[String: Any]] else {
+            throw NSError()
+        }
+        return dictionary
+    }
+    
+}
```

Comment:
Should be added in a separate file. 
- Recommended name: `Encodable+Helpers`
- Recommended Path: `Storage/Sources/Extensions/`